src/slurm_plugin/fleet_manager.py

```diff
@@ -243,24 +243,50 @@ def __init__(
             launch_overrides,
         )
 
+    def _evaluate_template_overrides(self) -> list:
+        """Build and return the list of Launch Template Overrides to be applied in the CreateFleet request.
+
+        (https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_FleetLaunchTemplateOverridesRequest.html)
+        """
+        template_overrides = []
+        overrides = {}
+
+        if self._compute_resource_config["CapacityType"] == "spot":
+            if self._compute_resource_config.get("MaxPrice"):
+                overrides = {"MaxPrice": str(self._compute_resource_config["MaxPrice"])}
```

Comment:
minor - if this code block is moved, than the assignment will break the logic. What about using ".update" also here?